### DISCUSSION TITLE
FluentDataGrid,  use byte[] with PropertyColumn

### DISCUSSION THREAD

**Discussion by andrewwang1tw** [microsoft/fluentui-blazor#2440] [Q&A] [answered]
Does FluentDataGrid support img from byte[] ? 

Content1 is the byte[] retrieved from database.

Below code only shows "System.Byte[]" on FluentDataGrid , not the really content

```
<PropertyColumn Property="@(c => c!.Content1)" />
```

```
<TemplateColumn Title="Picture" >
    <div class="flex items-center">
        <img class="flag" src="data:image/jpg;base64, @(context.Content1)" />
        <nobr>
            <strong>@context.Name</strong>
        </nobr>
    </div>
</TemplateColumn>
```

The result is as attached file

![未命名](https://github.com/user-attachments/assets/d9f910e6-2c5c-4a89-8600-971209c56c83)

**Comment by vnbaaij** [comment]
I'm guessing you need to do something with encoding/decoding. Can you display the images in plain html like in an unordered list?

**Reply by andrewwang1tw** [reply]
Will try it later, but below code shows "System.Byte[]". It seems the first problem. 

```
<PropertyColumn Property="@(c => c!.Content1)" />
```

**Reply by vnbaaij** [reply, marked as answer]
What else would you expect that to display? The property column uses `ToString()` method internally and for this array that will return the type/. You will need to do some work there to get it to display the value.

**Reply by andrewwang1tw** [reply]
the byte[] from database look like 

![image](https://github.com/user-attachments/assets/db116abd-3d7d-4763-957d-cbf44bb808a2)

Finally, the content is ok but the <img src=> still not work

![image](https://github.com/user-attachments/assets/68dc53c9-00c3-4069-b919-5ba6faeaf551)

**Reply by andrewwang1tw** [reply]
Finally, it works...

![image](https://github.com/user-attachments/assets/bc6ef551-1cf8-4e21-ab75-2dc71b01ea24)

**Reply by vnbaaij** [reply]
Can you post the solution here (for the good of the community). Thanks

**Reply by andrewwang1tw** [reply]
Here is the solution...

1. database side, the data type is byte[]

2. Web API side, I convert it to Base 64 string as below
```
"data:image/jpg;base64," + Convert.ToBase64String(c.Content1!) 
```

3. FluentDataGrid side, I use the "remote data" sample, we already have the base64 string, just set it on attribute "src" of "IMG" tag

```
 <TemplateColumn Title="Picture" >
     <div class="flex items-center">
         <img src="@context.Content1" />
```

The weird thing is if you use the byte[] data directly, the result will be the data type not the data content. I do not know why ?

It is just as you said.  Needs to encode first (converted and return the base64 "**string**" not return the byte[])

![image](https://github.com/user-attachments/assets/a84e27a6-4a72-4f0e-b121-a6a3c94cafcb)
